sklearn/tests/test_base.py

```diff
@@ -1000,3 +1000,79 @@ def test_get_params_html():
 
     assert est._get_params_html() == {"l1": 0, "empty": "test"}
     assert est._get_params_html().non_default == ("empty",)
+
+
+def make_estimator_with_param(default_value):
+    class DynamicEstimator(BaseEstimator):
+        def __init__(self, param=default_value):
+            self.param = param
+
+    return DynamicEstimator
+
+
+@pytest.mark.parametrize(
+    "default_value, test_value",
+    [
+        ((), (1,)),
+        ((), [1]),
+        ((), np.array([1])),
+        ((1, 2), (3, 4)),
+        ((1, 2), [3, 4]),
+        ((1, 2), np.array([3, 4])),
+        (None, 1),
+        (None, []),
+        (None, lambda x: x),
+        (np.nan, 1.0),
+        (np.nan, np.array([np.nan])),
+        ("abc", "def"),
+        ("abc", ["abc"]),
+        (True, False),
+        (1, 2),
+        (1, [1]),
+        (1, np.array([1])),
+        (1.0, 2.0),
+        (1.0, [1.0]),
+        (1.0, np.array([1.0])),
+        ([1, 2], [3]),
+        (np.array([1]), [2, 3]),
```

Comment:
@adrinjalali it is what you would expect? (The test is passing by the way).

```suggestion
        (np.array([1]), [2, 3]),
        (None, KFold()),
        (None, get_scorer("accuracy")),
```

Reply:
Yes, that test makes sense.